pandas/_libs/hashtable_func_helper.pxi.in

```diff
@@ -84,7 +84,7 @@ cpdef value_count_{{dtype}}({{c_type}}[:] values, bint dropna):
         int64_t[:] result_counts
         {{endif}}
 
-        Py_ssize_t k
+        khiter_t k
```

Comment:
this should be harmless, maybe a comment on the `for k in range(...)` explaining why we cant/dont use a Py_ssize_t there